### PR TITLE
cf-600E hehehehaw

### PR DESCRIPTION
**Pull request by envyaims**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: solutions/platinum/cf-600E.mdx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+const int MAXN = 1e5 + 1;
+array<ll, MAXN> color, ans;
```
Comment (SansPapyrus683) [inline solutions/platinum/cf-600E.mdx:25]: separate lines

File: solutions/platinum/cf-600E.mdx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+vector<int> g[MAXN];
+
+// col_cnt[i][j] = number of occurrences of color j in i
+map<ll, ll> col_cnt[MAXN];
```
Comment (SansPapyrus683) [inline solutions/platinum/cf-600E.mdx:29]: rename `col_occs` or smth

File: solutions/platinum/cf-600E.mdx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+const int MAXN = 1e5 + 1;
```
Comment (SansPapyrus683) [inline solutions/platinum/cf-600E.mdx:24]: 0-index the nodes

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
you know what i want

File: solutions/platinum/cf-600E.mdx (hunk 3 above)
Comment (envyaims) [reply solutions/platinum/cf-600E.mdx:24]: no

Comment (SansPapyrus683) [reply solutions/platinum/cf-600E.mdx:24]: in that case, i cannot in good conscience merge this pr

Comment (envyaims) [reply solutions/platinum/cf-600E.mdx]: this is whyt youre not liked

File: solutions/platinum/cf-600E.mdx (hunk 1 above)
Comment (envyaims) [reply solutions/platinum/cf-600E.mdx:25]: have you never defined arrays on the same line in your life

File: solutions/platinum/cf-600E.mdx (hunk 2 above)
Comment (envyaims) [reply solutions/platinum/cf-600E.mdx:29]: same thing???????

File: solutions/platinum/cf-600E.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/platinum/cf-600E.mdx]: :sob:

File: solutions/platinum/cf-600E.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/platinum/cf-600E.mdx:30]: then expand it to `col_count`

File: solutions/platinum/cf-600E.mdx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+int main() {
+	int n;
+	cin >> n;
+	for (int i = 0; i < n; i++) { cin >> color[i]; }
+	for (int i = 0; i < n - 1; i++) {
+		int u, v;
+		cin >> u >> v;
+		--u;
+		--v;
+		g[u].push_back(v);
```
Comment (SansPapyrus683) [inline solutions/platinum/cf-600E.mdx:72]:
```suggestion
		g[--u].push_back(--v);
```

File: solutions/platinum/cf-600E.mdx (hunk 2 above)
Comment (envyaims) [reply solutions/platinum/cf-600E.mdx:30]: same thing??
what else could cnt stand for?

Comment (SansPapyrus683) [reply solutions/platinum/cf-600E.mdx:30]: we had this conversation before

Comment (envyaims) [reply solutions/platinum/cf-600E.mdx:30]: and..?

Comment (SansPapyrus683) [reply solutions/platinum/cf-600E.mdx:30]: you agreed to rename it

Comment (envyaims) [reply solutions/platinum/cf-600E.mdx:30]: can you inform me why you want to rename it again?

File: solutions/platinum/cf-600E.mdx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+void dfs(int node, int p) {
+	// initialize this node
+	col_cnt[node][color[node]] += 1;
+	sum_occur[node][1] += color[node];
+
+	for (int i : g[node]) {
+		if (i == p) { continue; }
+		dfs(i, node);
+		// retrieve the biggest map so we have less to merge
```
Comment (ryanchou-dev) [inline solutions/platinum/cf-600E.mdx:42]:
```suggestion
		dfs(i, node);
		
		// retrieve the biggest map so we have less to merge
```

**Review by ryanchou-dev — APPROVED** [review]

File: solutions/platinum/cf-600E.mdx (hunk 5 above)
Comment (ryanchou-dev) [reply solutions/platinum/cf-600E.mdx:42]: js to space blocks of code out (optional)

File: solutions/platinum/cf-600E.mdx (hunk 2 above)
Comment (HarryWangATX) [reply solutions/platinum/cf-600E.mdx:30]: are you guys really arguing about a name change LOL

**Comment by HarryWangATX** [comment]
why dis not deploying :/

**Comment by SansPapyrus683** [comment]
image
💀

**Comment by SansPapyrus683** [comment]
image

bro what

**Comment by HarryWangATX** [comment]
i dont see anything wrong :skull:

**Comment by SansPapyrus683** [comment]
lmao can you just remake the pr, this one's kinda screwed

**Comment by envyaims** [comment]
> lmao can you just remake the pr, this one's kinda screwed

after u merge the other one cuz my branches are kinda screwed

**Pull request closed without merging**